docs/sql-pyspark-comparison-with-pandas.md

```diff
@@ -0,0 +1,401 @@
+
+PySpark does not allow the direct assignment whereas Pandas does.
+
+```python
+# Pandas
+df.city = 1
+```
+
+```
+   city  rank
+0     1     1
+1     1     2
+2     1     3
+```
+
+In PySpark, it should be done as:
+
+```
+df.withColumn("city", lit(1)).show()
+```
+
+This is fundamentally because PySpark DataFrame is immutable whereas Pandas DataFrame is mutable.
+
+
+### `NULL`, `None`, `NaN` and `NaT` 
+
+`NULL`, `None`, `NaN` and `NaT` concepts sometimes bring confusions to users and developers. Here are definitions of each:
+
+- In Spark SQL, `NULL` represents a missing value.
+- In Python, `None` represents a missing value.
+- In Pandas, for float types, `NaN` represents not-a-number and also a missing value. 
+- In Pandas, for `datetime64[ns]` type, `NaT` represents a missing value.
+
+For `NaN` comparison, sometimes confusions happen. Usually the comparison is not only specific in Pandas but also regular Python, NumPy, etc. 
+
+See the comparisons below:
+
+```python
+>>> np.nan and True
+True
+>>> True and np.nan
+nan
+```
+
+```python
+>>> np.nan and False
+False
+>>> False and np.nan
+False
+```
+
+```python
+>>> np.nan or True
+nan
+>>> True or np.nan
+True
+```
+
+```python
+>>> np.nan or False
+nan
+>>> False or np.nan
+nan
+```
+
+Python logical operators are different from SQL and some other programming languages. These comparisons are all correct as specified rules within Python. 
+
+**Note:** Python's boolean operator is different:
+
+| Operation   | Result                        |
+| ----------- | ----------------------------- |
+| `x or y	`   | if x is false, then y, else x |
+| `x and y`   | if x is false, then x, else y |
+
+See also [Boolean Operations](https://docs.python.org/3/library/stdtypes.html?highlight=short%20circuit#boolean-operations-and-or-not) in Python. 
+
+**Note:** `bool(np.nan)` is `True`. See also [Truth Value Testing](https://docs.python.org/3/library/stdtypes.html#truth-value-testing).
+
+**Note:** When you switch `np.nan` to `float("nan")`, it still shows the same results. Even when you switch `np.nan` to `np.datetime64('NaT')`, it also shows the same pattern (although the result `nan` becomes `np.datetime64('NaT')` instead).
+
+In PySpark, `None` comparison follows `NULL` comparison in Spark SQL.
+
+```
++-----+-----+---------+--------+
+|    x|    y|(x AND y)|(x OR y)|
++-----+-----+---------+--------+
+| null| true|     null|    true|
+| true| null|     null|    true|
+| null|false|    false|    null|
+|false| null|    false|    null|
++-----+-----+---------+--------+
+```
+
+`NaN` and `boolean` type comparison is disallowed.
+
+
+### Type inference, coercion and cast
+
+PySpark support native Python types currently. Those Python types are mapped to Spark SQL types in order to leverage Spark SQL. So, in general PySpark behaviours follow Spark SQL's behaviours whereas Pandas tend to look after Python's own behaviours and NumPy's behaviours since both types are supported in Pandas.
```

Comment:
support -> supports. 
To be consistent, behaviour -> behavior (UK vs US spelling)
tend -> tends
look after -> follow

Reply:
My english is completely mixed up between US and UK :) .. let me fix it.